recipe/2024-02-terraform-providers.md

```diff
@@ -0,0 +1,401 @@
++              subscriptionid: 1234,
++              secrets: {
++                source: secretStoreAz.id
++              }
++            }
++          },
++          {
++            name: 'azurerm',
++            properties: {
++              subscriptionid: 1234,
++              alias: 'az-paymentservice',
++              secrets: {
++                source: secretStoreAzPayment.id
++              }
++            }
++          },
++          {
++            name: 'gcp',
++            properties: {
++              project: 1234,
++              regions: ['us-east1', 'us-west1']
++            }
++          },
++          {
++            name: 'oraclepass',
++            properties: {
++              database_endpoint: "...",
++              java_endpoint: "...",
++              mysql_endpoint: "..."
++            }
++          }
++        ]
++      }
++      env: {
++        key: value
++        envSecrets: [                    // Individual Secrets from SecretStore
++          {
++             name: 'SPECIAL_LEVEL_KEY'
++             valueFrom: {
++               source: secretStoreConfig.id
++               name: 'special-config'
++               key: 'special.level'
++             }
++          },
++          {
++             name: 'LOG_LEVEL'
++             valueFrom: {
++               source: secretStoreConfig.id
++               name: 'special-config'
++               key: 'special.log'
++             }
++          }
++        ]
++        envSecretsFrom: {               // All Secrets from Secret Store
++           source: secretStoreOCICred.id
++         }        
++      }      
++    }
+  }
+  recipes: {      
+    ...
+  }
+}
+
+Option 2:
+This option provides grouping and efficiency to retrieve all details for a single provider.
+
+resource env 'Applications.Core/environments@2023-10-01-preview' = {
+  name: 'dsrp-resources-env-recipes-context-env'
+  location: 'global'
+  properties: {
+    compute: {
+      ...
+    }
+    providers: {
+      ...
+    }
+    recipeConfig: {
+      terraform: {
+        authentication:{
+          ...        
+        }
++       providers: {
++         'azurerm': [
++           {
++             subscriptionid: 1234,
++             secrets: {
++               source: 'secretStoreAz.id'
++            }
++          },
++          {
++             subscriptionid: 1234,
++             alias: 'az-paymentservice',
++             secrets: {
++               source: 'secretStoreAzPayment.id'
++            }
++          }]
++          'gcp': [
++            {
++              project: 1234,
++              regions: ['us-east1', 'us-west1']
++            }
++          ]
++          'oraclepass': [
++            {
++              database_endpoint: "...",
++              java_endpoint: "...",
++              mysql_endpoint: "..."
++            }
++          ]
++        }
++     }
++     env: {
++       key: value
++       envSecrets: [                     // Individual Secrets from SecretStore
```

Comment:
Can it just be `Secrets`? `env` part is a given since it is inside the env block.

```suggestion
+       secrets: [                     // Individual Secrets from SecretStore
```